main/src/main/scala/sbt/Defaults.scala

```diff
@@ -990,7 +990,14 @@ object Defaults extends BuildCommon {
     discoveredSbtPlugins := discoverSbtPluginNames.value,
     // This fork options, scoped to the configuration is used for tests
     forkOptions := forkOptionsTask.value,
-    selectMainClass := mainClass.value orElse askForMainClass(discoveredMainClasses.value),
+    allowUndiscoveredMainClass :== false,
```

Comment:
https://www.scala-sbt.org/1.x/docs/Plugins-Best-Practices.html#Provide+default+values+in
Could you move the default value to somewhere globally scoped so

```scala
ThisBuild / allowUndiscoveredMainClass := true 
```

would work?